### ISSUE TITLE
Is this possible to obfuscate "hover:"  class name or something like that from TailwindCSS? I also have some question and feature request

### ISSUE THREAD

**Issue by hoangnhan2ka3** [soranoo/next-css-obfuscator#6]
### **First of all, I really appreciate your efforts on this project ✨, thank you @soranoo and all Sponsors !**
This is almost what I needed, really easy to use, really full features and highly customizable.
Besides, it also helps my code look mysterious 😈 and professional.

I want to make some contributions to make this project even more perfect, to be widely known and to be a perfect replacement for a certain "no longer maintained package" named Post*** 😄.

My specs:

> OS: Windows 11
> IDE: VS Code v1.85.2
> Nextjs 14.1
> Frontend cloud: [Vercel](vercel.com)
> Using: [next-css-obfuscator](https://github.com/soranoo/next-css-obfuscator) v2.0.6

### **👉 1.  Also obfuscate something like ["hover:"](https://tailwindcss.com/docs/hover-focus-and-other-states) in TailwindCSS class:**

There are things that are really easy to confuse like `*:hover` in normal CSS and `hover:*` in TailwindCSS class, both have `hover` and `:`.

In my test project, I have ONLY one class end with `"...text-yellow-300"` (I mean there is no other class named "yellow") like this:

<img width="414" alt="Screenshot 2024-01-29 070831" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/6e56a3ea-12c0-4e4f-909b-123bdcc6cd7b">

&#32;
And when `npm run build`, in `css-obfuscator/conversion.json` file:

<img width="446" alt="Screenshot 2024-01-29 071613" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/f03f0e15-e1a9-46ad-bb35-219ca3f6a250">

_(I also delete `.next/cache` folder and old `css-obfuscator` folder, which contain `conversion.json` before running this build command)._

Result:

<img width="485" alt="Screenshot 2024-01-29 092640" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/e9aa1761-07bc-498e-8d5e-c05638f83270">

&#32;
**In expected:**

```
".hover\\:text-yellow-300": ".kj33u9",

// or at least

".hover\\:text-yellow-300": ".hover\\:kj33u9",
```

But no. So... I don't know what is the `:hover` suffix **(which is the main problem)** when detecting that class name 🥲. And why it detect another alone class `".text-yellow-300"` which doesn't exist.

Actually, I found a temporary solution when editing your `utils.js` file in the `node_modules` folder, I will talk about this problem in part 2.
<br>
### **👉 2.  The problem of** `selectorClasses.length > 1`:

I think part of the problem lies in this line of `utils.js`:
https://github.com/soranoo/next-css-obfuscator/blob/480fbb19b276b629b7eaf9df116fc4fbb65c886e/src/utils.ts#L801

When I edit it to `selectorClasses.length >= 1`, this is the result:

<img width="335" alt="Screenshot 2024-01-29 081320" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/3522d29e-f1b7-4582-910b-0a83788c6b34">

"Nears". But of course this does not work ⛔ too because of wrong class detected with `:hover` suffix.

Doesn't stop there, this edit also affects "single character" then "-[${custom parameter}]"
_eg: `z-[9999]`, `min-h-[80rem]`, etc. (almost all)_

&#32;
With `selectorClasses.length > 1`:

<img width="895" alt="Screenshot 2024-01-29 092229" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/4b9058e7-d1b7-4a41-a84a-03c027f07267">

&#32;
With `selectorClasses.length >= 1`:

<img width="893" alt="Screenshot 2024-01-29 084939" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/0beb699d-1afb-4b6e-b0e3-df6bd6347196">

Almost every TW custom parameter class except `hover:` works perfectly (for me) with `selectorClasses.length >= 1`. And I `run dev`, `run build + run start`  or `deploy to Vercel` with no error (I haven't tried all the TW classes and don't know if there are any special cases or exceptions, but this works for me for now)

🫶 If possible, please update an option for this `selectorClassesLength` as a temporary solution for custom parameter TW class.
<br>
### **👉 3.  Truth about select direct child ["*:"](https://tailwindcss.com/docs/hover-focus-and-other-states#styling-direct-children):**

If u know u know:

<img width="755" alt="Screenshot 2024-01-29 094126" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/ecb56dab-10f3-43d7-838e-4862c4102bde">

Even though this `*:` class exists in the parent div, it really only has an effect on the children div. 

I'm really not sure what will happen if we successfully obfuscate this `*:` class ⁉️

<img width="230" alt="Screenshot 2024-01-29 094937" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/e3051e57-ddae-4270-b8ef-f0a6daf0b8d9">

I think it's possible but quite difficult to do hmmm...

The script is to just obfuscate the `.\\*\\:font-extrabold` part and I think

```
.jzn6lz > * {
    font-weight: 800;
}
```

will do the rest.

Okay and... the remain problem of `*:` is same same as `hover:`

<img width="363" alt="Screenshot 2024-01-29 095924" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/4db7ba1c-bd3b-4657-b7e5-7f48c5d9be0e">

It is true that it must be `.\\*\\:font-extrabold > *` to work but we don't need to detect the `> *` part but `.\\*\\:font-extrabold` right?

At last I do think that fix the `hover:` problem is much possible than `*:` haha 😄.
<br>
### **👉 4.  Feature request:**

<img width="799" alt="Screenshot 2024-01-29 101716" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/2686e554-8317-4406-abde-c0403ddd8d68">

You know 👉👈... I think this feature is not useless but it's really not as useful as doing the opposite. I mean the tag with marker class will not be obfuscated, it is more valuable to use than having to add a marker to each tag you want to obfuscate, but yes, don't delete the current option because it is usable in some cases. Is it possible?

**Comment by hoangnhan2ka3** [comment]
Oh, I forgot, I was very surprised that when using `selectorClasses.length >= 1`, the class `.animate-\\[pulse-404_3s_linear_infinite\\]` becomes a super concise class like `.w2zhoa` but still used okay haha, that's the unexpected effect of this package.

**Comment by soranoo** [comment]
What a fantastic issue format! I really appreciate your enthusiasm and contribution to this package.

1. Regarding the issues, I will investigate them ASAP.
2. Regarding the opposite of obfuscate marker, I don't think it is a good idea to implement it right now. There is serious issue (see the [🐛 Known Issues](https://github.com/soranoo/next-css-obfuscator?tab=readme-ov-file#-known-issues)) with the current obfuscate marker feature and I don't think I can solve it shortly. The bigger obstacle at this moment is the way to track functions being called within a component so that we can obstacles its children's component. I tried to parse the JS and have some proof of concepts but I don't think I have enough knowledge handling this. If you are familiar with it please, please give me some hints.

**Comment by soranoo** [comment]
Plus, if you have a test project for this, would you mind to share with me?

**Comment by hoangnhan2ka3** [comment]
Thank you for reply!

1. Actually, I've been intending to obfuscate this TW class for a long time. I only see [facebook.com](https://facebook.com) and [blueagle.top](https://blueagle.top/) available. like this, but it seems that the two still cannot agree on the number of characters in the obfuscated class, which is still not perfect (unlike your package 💕).

2. Yeah you're right, the issue of `opposite markers` is probably not the highest priority right now. If feasible, can you address the fixable issues first? I mean the `selectorClasses.length >= 1` problem.

3. Regarding testing project, I will comment here ASAP when I finish preparing. (Again, sorry that's as far as I can help right now).
This is [my test domain](https://hoangnhan.uk), you can check out. Try [the 404 page](https://hoangnhan.uk/something) with mouse wheel scroll 😄.

5. I actually have more than one feature request 😄

- Like being able to identify and replace the class `.big` in `const bigElement = document.querySelector(".big")`
- A feature to delete the old TW class because really after If there is a new obfuscated class, the old class is no longer needed, and if it exists, it will really increase the weight of the css file. It might be really difficult but try reversing the way you detect the classes before obfuscating and saving, then after everything is done, delete it.
&#32;
_(I'm not sure if after running this package, new classes are created. Did it release at once and then add it with TW's css file? Or the obfuscate process is done "in parallel" with TW's css file export process, because I see each new obfuscated class coming next to a class with similar functionality but of TW. Rather than the entire new obfuscated class being located at the bottom of the css file or separated into another file)._

If you can really do this, this will be a huge boom for your package.

#waitingforamiracle

**Comment by soranoo** [comment]
Yes, fixing the bugs is my top priority right now.

Regarding the feature request,
1. So you would like to replace a specific class with a class you want?
2. Maybe I should add an option to del all TW classes if they have an obfuscated version. The reason I keep the TW classes is to prevent breaking the site when some obfuscation fails.

**Comment by hoangnhan2ka3** [comment]
Yeah if you can actually delete exactly the OBFUSCATED TW classes then maybe there won't be any problem, otherwise you can try making a beta version of the package for example.

**Comment by soranoo** [comment]
@hoangnhan2ka3 The beta version is really! 

> npm run next-css-obfuscator@beta

Looking forward to your feedback~

**Comment by hoangnhan2ka3** [comment]
hi there @soranoo =))))

This my first look about the fix of `*:` class

<img width="841" alt="Screenshot 2024-01-31 084943" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/49ba635d-85c9-4339-be95-55c83ecfc2de">

Wait for me for full feedback, looks very interesting 💣

**Comment by soranoo** [comment]
### Investigate Report
1. `hover:` and `:hover` 
   - Since I have used Tailwind for quite a long time I had forgotten the `hover:` is from Tailwind. 🙃
   - Incorrect pseudo-classes handling. https://github.com/soranoo/next-css-obfuscator/commit/bb90dd0f259a9aa17e77d427f43fcac246a5c760
2. The problem of `selectorClasses.length > 1`  and Tailwind universal selector
   -  Due to some defect of class extraction regex. https://github.com/soranoo/next-css-obfuscator/commit/bb90dd0f259a9aa17e77d427f43fcac246a5c760 https://github.com/soranoo/next-css-obfuscator/commit/ce8d7831f2f550db92e939308b2f4f733d262af8

### Feature Update
- Removed `customTailwindDarkModeSelector` option
- Delete original CSS after obfuscation automatically (for full obfuscation)

New [REAMME](https://github.com/soranoo/next-css-obfuscator/blob/development/README.md)

**Comment by hoangnhan2ka3** [comment]
I'm extremely honored to be tagged in the README ✨.

And I'm surprised by the structure of your code after this change quite a bit, thank you for your effective efforts 🎉.

There are actually a lot of improvements, especially with the `hover:` class, which is almost fine for now.

You also fixed the problem https://github.com/soranoo/next-css-obfuscator/commit/bb90dd0f259a9aa17e77d427f43fcac246a5c760#commitcomment-138032346

<img width="272" alt="Screenshot 2024-01-31 115010" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/764eb7c8-6157-44e2-95ab-10d9025b4a28">

which is noice 💕💕.

Ok now let's get to the main point.

### **👉 1. The Biggest One `*:` class:**

> hi there @soranoo =))))
> 
> This my first look about the fix of `*:` class
> 
> <img alt="Screenshot 2024-01-31 084943" width="841" src="https://private-user-images.githubusercontent.com/147973044/301013109-49ba635d-85c9-4339-be95-55c83ecfc2de.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY2NjY1MzIsIm5iZiI6MTcwNjY2NjIzMiwicGF0aCI6Ii8xNDc5NzMwNDQvMzAxMDEzMTA5LTQ5YmE2MzVkLTg1YzktNDMzOS1iZTk1LTU1YzgzZWNmYzJkZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEzMVQwMTU3MTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wYzc4YzdmOTQ4Y2FiOTU4MDgzMTBkZTViYmY1YWFmNDRmOWE1YTIyNWE3ODQzY2ZjMGMwMTZkZGU3YmUxMTBkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.7AMGo_HZp8s8V_Hb_16A9eqmXiQXwr0b-W-dmKFK22s">
> Wait for me for full feedback, looks very interesting 💣

Well~, the problem is the same of `hover:` class I've commented on your edit https://github.com/soranoo/next-css-obfuscator/commit/bb90dd0f259a9aa17e77d427f43fcac246a5c760#commitcomment-138032286 yesterday:

+ Obfuscated ✅:

<img width="307" alt="Screenshot 2024-01-31 100206" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/ad640dda-8c21-4f92-ae99-7ccd8835b412">

Good group separation !

+ Replaced ✅:

<img width="443" alt="Screenshot 2024-01-31 100933" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/69d48a4c-7682-4f85-b105-b7c11f8a9430">

&#32;
<img width="608" alt="Screenshot 2024-01-31 102537" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/f917ab3b-b112-41f1-a90e-f25a52eee8f8">

Yeah I mean **"replace"**, not **"change"**:
```
/* From this */
.\*\:justify-center > * {
    justify-content: center;
}

/* To this */
.\*\:hbk62l > * {
    justify-content: center;
}
```

**IN THE CSS FILE !!!** 

_Or at least for now, it can be called **"add more"**.
Because the problem I'm about to talk about in next part still doesn't seem to be quite right, even though you mentioned it in the Change log._

+ Doesn't have an effect cuz there are no `.hbk62l` in css file ⛔:

<img width="277" alt="Screenshot 2024-01-31 103556" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/42f6b1e7-15ec-4132-b271-32f603fe8f9f">

I also tried deploying to Vercel, Redeploy without caching,... but nothing changed...

### **👉 2. Is this real 🤔:**

In new beta Readme, you said `Delete original CSS automatically after obfuscation (only apply at full obfuscation)`, WOW 🤯.

After reading carefully the new `Tips` section, I think it's an issue.

Looks like I haven't experienced that yet, don't know if it's a problem because I'm using `tsx` or if our `version` is different.
As u can see, unused classes have not been deleted/ replaced. (I tried with one default and one custom TW class but same output parameter)

<img width="1220" alt="Screenshot 2024-01-31 105911" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/07090c5c-cdb2-4fe1-be69-575970323c42">

Easy-to-see version of View page source:

<img width="178" alt="Screenshot 2024-01-31 110951" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/227e39ae-e817-4780-bde7-692672b700e3">

In `conversion.json` file:

<img width="261" alt="Screenshot 2024-01-31 110821" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/6c50f3a1-e714-4f6f-b767-d22eb938b299">

But please don't worry, as I said before, let's solve the superficial problems first, a little file size of css file is not a big deal now, but it would be a really great option if it could be done.

### **I was planning to suggest more features but it doesn't seem necessary right now, I'm looking forward to your next patch @soranoo .**

In addition, I also discovered some more bugs:

+ Although these classes have been included in `classIgnore`, perhaps this update makes it no longer work.

<img width="366" alt="Screenshot 2024-01-31 111813" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/3e23faa5-c53f-46e1-91fd-8c0d7f0e06ea">

In my `next-css-obfuscator.config.cjs`:

<img width="466" alt="Screenshot 2024-01-31 112404" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/e7a136e3-6969-4e2f-9c39-45a192e2afec">

+ Select/ detect class `without space` with attribute:

<img width="587" alt="Screenshot 2024-01-31 114415" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/50cacff7-2464-4e24-8862-26f294ca5a88">

Yeah this `[pop-t]`, `[pop-b]`, `[pop-l]` and `[pop-r]` is my custom attribute used for long-hover popup like this:

<img width="353" alt="Screenshot 2024-01-31 113209" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/0e9ffeb3-2258-49fb-87ff-1f648a0a6666">

<img width="422" alt="Screenshot 2024-01-31 113314" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/72714454-1888-4aa7-80c7-82458538a7ca">

Just like [title](https://www.w3schools.com/tags/att_global_title.asp) attribute:

<img width="295" alt="Screenshot 2024-01-31 113506" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/05cace6f-2067-4bdc-83e8-73faf69a62c3">

The project containing `pop` is an old project based on static HTML so I haven't tried it on the Nextjs project this time, but I think it will also have some problem with deleting/ replacing the obfuscated class inside. css files

**Comment by hoangnhan2ka3** [comment]
It seems more correct then ~ @soranoo 

Before:

<img width="584" alt="Screenshot 2024-01-31 110318" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/e089a364-4181-4d9b-a964-263ccdacfff0">

After:

<img width="599" alt="Screenshot 2024-01-31 194007" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/72790cc3-9af3-4651-aad2-866f489683c0">

✨ I don't see classes that should not be duplicated anymore.

But what happen with the `new CSS rules count` haha 🤔.

**Comment by soranoo** [comment]
### Tailwind Universal Selector
- Regarding the Tailwind universal selector, `*\:class` is a single class (like `dark\:class` and `hover\:class`) so there is no reason to treat it differently.

### `classIgnore`
- The `classIgnore` is expected to only accept `string[ ]` therefore `RegExp` doesn't work with it. (I will support it later on)
- I don't see any issue with `classIgnore`. But note that `.hex.hex-c` are two classes, you should register them as `["hex", "hex-c"]` not `["hex.hex-c"]`

### [attribute] Selector
- Fixed https://github.com/soranoo/next-css-obfuscator/commit/80aacb7eb5f9fc639cf4b4c286fc69c4ef4150e3

**Comment by hoangnhan2ka3** [comment]
✨Report to commander @soranoo , mission success !

<img width="758" alt="Screenshot 2024-01-31 224212" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/85dd8847-90ae-4983-8e0e-8c4472cfc199">

<img width="193" alt="Screenshot 2024-01-31 223634" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/aec0066e-7e26-493a-80a0-05e2f534bc58">

<img width="523" alt="Screenshot 2024-01-31 224055" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/11257110-3b48-4adc-9d02-16a2abd4ed0c">

I've been trying with `new CSS rules count` but it still doesn't work 🥲
Please provide an official version to npm.

<img width="574" alt="Screenshot 2024-01-31 223801" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/579bb672-e73d-417b-a68f-413161e1785e">

**Comment by hoangnhan2ka3** [comment]
Plus, there are 2 `Obfuscation` in the log 👀:

<img width="245" alt="Screenshot 2024-01-31 231643" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/e94bfd21-1cd0-48c2-943d-a6770ffb07d3">

**Comment by soranoo** [comment]
Let's have a final beta before release! 

> npm run next-css-obfuscator@beta

I had added/merged some options. Btw `classIgnore` option now supports Regex.
[New README](https://github.com/soranoo/next-css-obfuscator/blob/development/README.md)

**Comment by soranoo** [comment]
> Plus, there are 2 `Obfuscation` in the log 👀:
> 
> <img alt="Screenshot 2024-01-31 231643" width="245" src="https://private-user-images.githubusercontent.com/147973044/301230783-e94bfd21-1cd0-48c2-943d-a6770ffb07d3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY3MTgzMDAsIm5iZiI6MTcwNjcxODAwMCwicGF0aCI6Ii8xNDc5NzMwNDQvMzAxMjMwNzgzLWU5NGJmZDIxLTFjZDAtNDhjMi05NDNkLWE2NzcwZmZiMDdkMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEzMVQxNjIwMDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01MTlkZmQ0MTM0MWM1Y2E5YWRjZjAwZTJiNTRmNzAxNTk2NjE5YThiOWI0ZTRiMzY3ZDkzMWUwOWZjZGEwMGM2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.EBPodFjOVYNBCJhxx8cwgCaY_g_Fdv2wxwi9NfPjvdg">

I will remove it before the official release.

**Comment by hoangnhan2ka3** [comment]
Okayyyy newwww buggggg:

```
// layout.tsx

import localFont from 'next/font/local';

const gilroy = localFont ({
    src: [
        {
            path: "../fonts/SVN-Gilroy.woff2",
            weight: "400",
            style: "normal",
        },
        {
            path: "../fonts/SVN-GilroyMedium.woff2",
            weight: "500",
            style: "normal",
        },
        {
            path: "../fonts/SVN-GilroySemiBold.woff2",
            weight: "600",
            style: "normal",
        },
        {
            path: "../fonts/SVN-GilroyBold.woff2",
            weight: "700",
            style: "normal",
        },
        {
            path: "../fonts/SVN-GilroyHeavy.woff2",
            weight: "800",
            style: "normal",
        },
        {
            path: "../fonts/SVN-GilroyBlack.woff2",
            weight: "900",
            style: "normal",
        },
    ],
});

export default function RootLayout({
    children,
}: Readonly<{
    children: React.ReactNode;
}>) {
    return (
        <html lang="en" className={gilroy.className}>
            <head>
                <meta httpEquiv="X-UA-Compatible" content="IE=edge,chrome=1" />
                <meta name="last-modified" content={new Date().toLocaleDateString()}/>
            </head>
            <body className="dark">
                <noscript>You need to enable JavaScript to run this site.</noscript>
                <ProgressTopbar />
                <main className="flex">
                    <HomeLayout />
                    <div className="p-6 size-full">
                        {children}
                    </div>
                </main>
            </body>
        </html>
    );
}
```

U see `<html lang="en" className={gilroy.className}>`

In production, it will look like this if no obfuscation:

<img width="293" alt="Screenshot 2024-01-31 233519" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/a8869656-d792-4ede-9c27-7e20f7d4b8ed">

Please refer to [Nextjs local fonts](https://nextjs.org/docs/app/building-your-application/optimizing/fonts#local-fonts)

I tried not including `__className_38b0b7` in `classIgnore` so it looks like a normal person.

And when run this package:

In normal pages ✅:

<img width="1280" alt="Screenshot 2024-01-31 233018" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/6ed280a2-bced-46ea-bb2d-80ce521abfe2">

In 404 page ⛔:
 
<img width="1280" alt="Screenshot 2024-01-31 234112" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/14cf76a8-d1e6-41c1-9a36-dce61f4ad72f">

Yes, fonts have not been applied.

In previous comments, I put `__className_38b0b7` in `classIgnore` so I didn't detect this problem.

`__className_38b0b7` will not change even when deploying or `run build + run start` or `run dev`, but when you add fonts or update fonts I think it will change. So I think adding exactly `__className_38b0b7` to `classIgnore` is not a good idea, right?

**Comment by soranoo** [comment]
try put `/__.*/` to `classIgnore `?

**Comment by hoangnhan2ka3** [comment]
Nope,

+ Case 1:

<img width="458" alt="Screenshot 2024-01-31 235339" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/bea5bfa9-c745-40ff-923c-297715f6ee27">

Result 1:

<img width="326" alt="Screenshot 2024-01-31 235142" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/826183c4-7eab-4c0a-9298-f05fbc7141f9">

+ Case 2:

<img width="475" alt="Screenshot 2024-01-31 235436" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/5d54f46a-e5f8-47fd-8601-0dc7c452a003">

Result 2:

<img width="305" alt="Screenshot 2024-01-31 235304" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/c54cba52-e797-498c-8fb1-ea12776a0d8e">

Do you have any updates about `classIgnore` that I missed? Do I need to update?

**Comment by hoangnhan2ka3** [comment]
okay I just updated with `classIgnore` which already supports `Regex`, everything is fine, so a class like `__className_38b0b7` cannot be obfuscated, right?

I also just tried with some other `Regex` and they work fine too:

<img width="445" alt="Screenshot 2024-02-01 002907" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/fe032e72-0487-4c7a-acfd-8322253aea74">

Result:

<img width="643" alt="Screenshot 2024-02-01 003140" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/c54402e8-79a6-4c4a-abca-936da94a03d1">

Thanks @soranoo 💕!

**Comment by soranoo** [comment]
I guess yes... I will add this to the doc

**Comment by hoangnhan2ka3** [comment]
So it seems like there's only one "problem" left, which is replacing the class in `querySelector` or something like that.

<img width="368" alt="Screenshot 2024-02-01 003650" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/06c40ee8-093e-4a82-811a-c74f882b5fba">

I tried removing `.big` and `.small` from `classIgnore` and the rotation effect no longer seems to work.

**Comment by hoangnhan2ka3** [comment]
One more :)))) Sorry I'm so annoying just pointing out the bug without doing anything to help you 🥲.

### **👉 `filter` tag inside `svg` tag:**

<img width="495" alt="Screenshot 2024-02-01 005032" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/bd148a48-bab6-4bba-bc9c-e46ecc505d65">

In the comments above, I had to put `filter` in `classIgnore` to not cause the `filter` tag to obfuscate, hmmm... I don't know why it detected `filter` tag as a class.

<img width="437" alt="Screenshot 2024-02-01 005433" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/358821da-aeee-4bba-b779-2e4d5b1c64c0">

In `conversion.json` file:

<img width="171" alt="Screenshot 2024-02-01 005736" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/5c65eef2-813e-4bf4-b155-086d985a42ea">

OMG is it really a class 🤯? How.

### **👉 Also with `iframe` tag:**

<img width="282" alt="Screenshot 2024-02-01 010014" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/3618b765-bc16-446f-b123-7f0fb89d4f21">

<img width="174" alt="Screenshot 2024-02-01 010221" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/29d0e2af-3d83-47f6-9cd9-c436213a41cf">

I thought it was an `element of HTML` huh?

<img width="504" alt="Screenshot 2024-02-01 010054" src="https://github.com/soranoo/next-css-obfuscator/assets/147973044/5791b8b6-a429-47f2-b7cd-37cb8e2ce815">

**Comment by soranoo** [comment]
> So it seems like there's only one "problem" left, which is replacing the class in `querySelector` or something like that.
> 
> <img alt="Screenshot 2024-02-01 003650" width="368" src="https://private-user-images.githubusercontent.com/147973044/301255146-06c40ee8-093e-4a82-811a-c74f882b5fba.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY3Mjg4ODAsIm5iZiI6MTcwNjcyODU4MCwicGF0aCI6Ii8xNDc5NzMwNDQvMzAxMjU1MTQ2LTA2YzQwZWU4LTA5M2UtNGE4Mi04MTFhLWM3NGY4ODJiNWZiYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEzMVQxOTE2MjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wNzRiZGQ4OTQ5ZjU5NzAxNTg4ZGUxZWE4ZDg3MGEzMDdmYWUwODlmZjY3YTA0MGVhN2M2M2U4YzUyMDNmMThmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.di-njXajiuZDmN-3IFqa6x7vm_gz2CJc1-swasWBvl4">
> I tried removing `.big` and `.small` from `classIgnore` and the rotation effect no longer seems to work.

I don't think is it a good idea to use `querySelector` but not `useRef`. You are using `React` not `jQuery`

**Comment by soranoo** [comment]
After some testing, I decided not to edit the regex since the javascript will messed up by that `.`.

**Comment by hoangnhan2ka3** [comment]
That's okay, don't try too hard

**Comment by hoangnhan2ka3** [comment]
That's not called a natural deployment, right?

Okay, I'll live with `.big` and `.small` in classIgnore, no problem 🤡

**Comment by soranoo** [comment]
Yes, directly editing the conversion JSON is not recommended. Btw can I mention your site in the dome section?

**Comment by hoangnhan2ka3** [comment]
Ok, but let me provide another domain with same content, cuz I will use hoangnhan.uk for more deployment.

U can use [hoangnhan.co.uk](https://hoangnhan.co.uk), I will update it in a minute

**Comment by hoangnhan2ka3** [comment]
[hoangnhan.co.uk](https://hoangnhan.co.uk) is ready @soranoo, I just wait for new npm package to update and u can mention it.

**Comment by soranoo** [comment]
Please notify me when your site goes into production! Looking forward to your fantastic work.

**Comment by hoangnhan2ka3** [comment]
It's just a 100dvh website haha 🤡

**Comment by soranoo** [comment]
@hoangnhan2ka3 The `2.1.0` is out
